### PR TITLE
increase test coverage, refactor duplicate code.

### PR DESCRIPTION
**Pull request by mikehelmick**
Towards #1279 

## Proposed Changes

* increase coverage on exportimpoert model + database
* increase coverage on json parsing
* collapse multiple copies of Public Key parsing function

Overall coverage increased by .7 percent w/ this change.

**Release Note**

```release-note
Increase test coverage on exportimport model and database packages.
```

### CODE REVIEW

File: internal/exportimport/database/exportimport.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ import (
 	"github.com/jackc/pgx/v4"
 )
 
+// ExportImportDB contains databse methods for managing with export-import configs.
```
Comment (whaught) [inline internal/exportimport/database/exportimport.go:29]:
```suggestion
// ExportImportDB contains database methods for managing with export-import configs.
```

File: internal/exportimport/model/exportimport.go [hunk 2]
```diff
@@ -20,6 +20,9 @@ import (
 	"time"
 )
 
+// ExportImport reqpresnts the configuration of a set of export files
```
Comment (whaught) [inline internal/exportimport/model/exportimport.go:23]:
```suggestion
// ExportImport represents the configuration of a set of export files
```

File: internal/exportimport/model/exportimport.go [hunk 3]
```diff
@@ -46,6 +51,8 @@ func (ei *ExportImport) Validate() error {
 	return nil
 }
 
+// Active returns if the ExportImport configuration is currently
```
Comment (whaught) [inline internal/exportimport/model/exportimport.go:54]:
```suggestion
// Active returns true if the ExportImport configuration is currently
```

File: internal/exportimport/model/exportimport_test.go [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+				IndexFile:  "a/index.txt",
+				ExportRoot: "a",
+			},
+			want: "",
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			got := ""
```
Comment (whaught) [inline internal/exportimport/model/exportimport_test.go:76]: can we t.Parallel the inner `t.Run` too in these tests?

File: internal/exportimport/model/importkeys.go [hunk 5]
```diff
@@ -16,13 +16,15 @@ package model
...
-	"errors"
-	"fmt"
 	"time"
+
+	"github.com/google/exposure-notifications-server/pkg/keys"
 )
 
+// ImportFilePublicKey represents a possible signing key
+// for export files being imported into this system.
+// A given ExportImportID can have more than one, and more than
```
Comment (whaught) [inline internal/exportimport/model/importkeys.go:26]:
```suggestion
// A given ExportImportID can have more than one associated key, and more than
// one key may be currently valid
```

File: internal/jsonutil/unmarshal_test.go [hunk 6]
```diff
@@ -28,6 +30,21 @@ import (
 	"github.com/google/go-cmp/cmp"
 )
 
+func TestBodyTooLarge(t *testing.T) {
```
Comment (whaught) [inline internal/jsonutil/unmarshal_test.go:33]: I think these can be t.Paralleled too?

**Review by whaught — APPROVED** [review]

File: internal/exportimport/database/exportimport.go [hunk 7]
```diff
@@ -71,7 +74,7 @@ func (db *ExportImportDB) ListConfigs(ctx context.Context) ([]*model.ExportImpor
 				id, index_file, export_root, region, from_timestamp, thru_timestamp
 			FROM
 				exportimport
-			ORDER BY id DESC
+			ORDER BY id ASC
```
Comment (sethvargo) [inline internal/exportimport/database/exportimport.go:77]: Hmm - this is a behavior change. Previously they were listed with the newest at the top. Now they'll be listed with the newest at the bottom. That feels wrong.

Comment (mikehelmick) [reply internal/exportimport/database/exportimport.go:77]: yes - this is for admin console display only.
In the only known production usage of this, there are exactly 3 entries in this table.

**Pull request merged**